pkg/manager/impl/execution_manager_test.go

```diff
@@ -2409,6 +2409,71 @@ func TestAssignResourcesIfUnset(t *testing.T) {
 			Value: taskResourceSpec.Memory,
 		},
 	}, assignedResources)
+
+	assert.EqualValues(t, taskResourceIndices{
+		memoryIndex: 1,
+		cpuIndex:    0,
+	}, indices)
+}
+
+func TestFinalizeTaskResources(t *testing.T) {
+	t.Run("LimitsLessThanRequests", func(t *testing.T) {
+		task := &core.CompiledTask{
+			Template: &core.TaskTemplate{
+				Target: &core.TaskTemplate_Container{
+					Container: &core.Container{
+						Resources: &core.Resources{
+							Requests: []*core.Resources_ResourceEntry{
+								{},
+								{
+									Value: "2",
+								},
+							},
+							Limits: []*core.Resources_ResourceEntry{
+								{
+									Value: "1",
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		var resourceRequestIndex, resourceLimitIndex = 1, 0
+		err := finalizeTaskResources(context.Background(), task, resourceRequestIndex, resourceLimitIndex, "my imaginary resource")
+		assert.Nil(t, err)
+		assert.EqualValues(t, task.Template.GetContainer().Resources.Requests[resourceRequestIndex].Value,
```

Comment:
In your function the logic is that when request > limit, you **lower** the value of request to the same value as limit. I think in this test that behavior should be stated more expliciltly.